TheMapCloudWebServices/layers.py

```diff
@@ -1,25 +1,174 @@
+                min = scale_hint.attrib.get('min')
+                max = scale_hint.attrib.get('max')
+
+            # Create layer dict entry
+            layer_metadata = {'title': title,
+                              'name': name,
+                              'min_scale': min,
+                              'max_scale': max}
+            layers.append(layer_metadata)
+
+        return layers
+
+    def extract_metadata_from_getcapabilities(self, xml_root):
+        """
+        Extract the metadata/contact info from the WMS
+        GetCapabilities XML.
+        :param xml_root: XML Element of GetCapabilities
+        :return: dict of metadata
+        """
+        service = xml_root.find('./Service')
+        contact_info = service.find('./ContactInformation')
+
+        metadata = {
+            'abstract': service.find('./Abstract').text,
+            'person': contact_info.find('./ContactPersonPrimary/ContactPerson').text,
+            'position': contact_info.find('./ContactPosition').text,
+            'org': contact_info.find('./ContactPersonPrimary/ContactOrganization').text,
+            'address': contact_info.find('./ContactAddress/Address').text,
+            'city': contact_info.find('./ContactAddress/City').text,
+            'country': contact_info.find('./ContactAddress/Country').text,
+            'postcode': contact_info.find('./ContactAddress/PostCode').text,
+            'phone': contact_info.find('./ContactVoiceTelephone').text,
+            # 'fax': contact_info.find('./ContactFacsimileTelephone').text,
+            'email': contact_info.find('./ContactElectronicMailAddress').text,
+            'access': service.find('./AccessConstraints').text
+        }
+        return metadata
+
+    def extract_wmts_layers_from_getcapabilities(self, xml_root):
+        """
+        Extract the WMTS layers from the WMTS
+        GetCapabilities XML.
+        :param xml_root: XML Element of GetCapabilities
+        :return: list of layer dicts
+        """
+        layers = []
+
+        all_layers = xml_root.findall('./{http://www.opengis.net/wmts/1.0}Contents/'
+                                      '{http://www.opengis.net/wmts/1.0}Layer')
+
+        for layer in all_layers:
+
+            # Create layer dict entry
+            title = layer.find('./{http://www.opengis.net/ows/1.1}Title').text
+            grid = layer.find('./{http://www.opengis.net/wmts/1.0}TileMatrixSetLink/'
+                              '{http://www.opengis.net/wmts/1.0}TileMatrixSet').text
+            name = layer.find('./{http://www.opengis.net/ows/1.1}Identifier').text
+
+            layer_metadata = {'title': title,
+                              'name': name,
+                              'grid': grid,
+                              'min_scale': None,
+                              'max_scale': None}
+            layers.append(layer_metadata)
+
+        return layers
 
 
 class PreviewButton(QStyledItemDelegate):
```

Comment:
I have removed some of the preview code, however, as I dont have the original UI files I cant remove the button easily. So I have had to leave some code in to disable on startup.